integrations/integration.php

```diff
@@ -135,18 +156,114 @@ public function get_slug(): string {
 	}
 
 	/**
-	 * Set vip_config property.
+	 * Set `vip_configs` property.
 	 *
-	 * @param IntegrationVipConfig $vip_config Instance of IntegrationVipConfig.
+	 * @param array<VipIntegrationConfig> $vip_configs Configurations provided by VIP.
 	 *
 	 * @return void
+	 *
+	 * @private
 	 */
-	public function set_vip_config( IntegrationVipConfig $vip_config ): void {
-		if ( ! $this->is_active() ) {
-			trigger_error( sprintf( 'Configuration info can only assigned if integration is active.' ), E_USER_WARNING ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
+	public function set_vip_configs( array $vip_configs ): void {
+		$this->vip_configs = $vip_configs;
+	}
+
+	/**
+	 * Get statuses of the integration in context of current site.
+	 *
+	 * @return array<Env_Integration_Status>
+	 */
+	private function get_site_statuses() {
+		/**
+		 * Statuses of the integration.
+		 *
+		 * @var array<string>
+		 */
+		$statuses = [];
+
+		foreach ( $this->vip_configs as $vip_config ) {
+			if ( $this->get_value_from_config( $vip_config, 'org', 'status' ) === Org_Integration_Status::BLOCKED ) {
+				return [ Org_Integration_Status::BLOCKED ];
+			}
```

Comment:
Can we instead make these decisions in goop before setting the configmap?

Anything that doesn't belong on a site config (based on inheritance from the client level) shouldn't end up in the configmap in the first place.

Reply:
1. Currently we can get the info about why a particular integration doesn't have any data in ConfigMap by just looking at the file. If we move it to GOOP then we loses this visibility which effect debugging if user doesn't have enough permissions.

2. In future we can have integrations that want to show some notification in WP Dashboard if the plugin is `blocked` on client level. So it's useful to have this info available.

If we decide on making any change then I will make these changes in the next PR because this PR is just doing refactoring.